File: src/test/functional/org/apache/prepbuddy/functionaltests/framework/DatasetTestResults.java
package org.apache.prepbuddy.functionaltests.framework;

public class DatasetTestResults {
    private String result;
    private String testSpecName;
    private AssertionError cause;

    public DatasetTestResults(String testSpecName) {
        this.testSpecName = testSpecName;
    }

    public void markAsSuccess() {
        result = "SUCCESSFUL";
    }

    public void markAsFailure(AssertionError error) {
        result = "FAILURE";
        cause = error;
    }

    public String getOutcome() {
        return result;
    }

    public String getCause() {
        return cause.getMessage();
    }
}

File: src/test/functionaltest/framework/DatasetTestResults.scala
package framework

class DatasetTestResults(testSpecName: String) {
    private var result: String = null
    private var cause: AssertionError = null

    def markAsSuccess() {
        result = "SUCCESSFUL"
    }

    def markAsFailure(error: AssertionError) {
        result = "FAILURE"
        cause = error
    }

    def getOutcome: String = result

    def getCause: String = cause.getMessage
}
